cmd/tsbs_generate_data/common/distribution.go

```diff
@@ -206,3 +206,36 @@ func FP(step Distribution, precision int) *FloatPrecision {
 		precision: math.Pow(10, float64(precision)),
 	}
 }
+
+
+// SaddledDistribution is a distribution that can change it's value
+// only if a "saddle" distribution provides a value above a specified threshold.
+// Otherwise it remains the same.
+type SaddledDistribution struct {
+	Saddle    Distribution
```

Comment:
I think all of these can be unexported. Don't want somebody accidentally messing with struct fields